lib/membrane_sdl/player.ex

```diff
@@ -45,23 +41,31 @@ defmodule Membrane.SDL.Player do
   end
 
   @impl true
-  def handle_start_of_stream(:input, ctx, state) do
-    use Numbers, overload_operators: true
-    {nom, denom} = ctx.pads.input.stream_format.framerate
-    timer = {:demand_timer, Ratio.new(Time.seconds(denom), nom)}
-
-    {[demand: :input, start_timer: timer], %{state | timer_started?: true}}
+  def handle_start_of_stream(:input, _ctx, state) do
+    {[demand: :input, start_timer: {:demand_timer, :no_interval}], state}
   end
 
   @impl true
-  def handle_buffer(:input, %Buffer{payload: payload}, _ctx, state) do
+  def handle_buffer(:input, %Buffer{payload: payload, pts: pts}, _ctx, state) do
     payload = Membrane.Payload.to_binary(payload)
-    :ok = CNode.call(state.cnode, :display_frame, [payload])
-    {[], state}
+
+    actions =
+      case state do
+        %{last_pts: nil, last_payload: nil} ->
+          [demand: :input]
+
+        %{last_pts: last_pts, last_payload: last_payload} ->
+          :ok = CNode.call(state.cnode, :display_frame, [last_payload])
+
+          [timer_interval: {:demand_timer, pts - last_pts}]
```

Comment:
- when we get the first frame, we should display it immediately
- when we get a subsequent frame, we should display it in `handle_tick`

so it looks like it should be the other way round